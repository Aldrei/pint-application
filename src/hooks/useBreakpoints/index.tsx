import { useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';

export const useBreakpoints = () => {
  const theme = useTheme();

  const goSm = useMediaQuery(theme.breakpoints.up('sm'));
  const goMd = useMediaQuery(theme.breakpoints.up('md'));
  const goLg = useMediaQuery(theme.breakpoints.up('lg'));
  const goXl = useMediaQuery(theme.breakpoints.up('xl'));

  return [goSm, goMd, goLg, goXl];
};
