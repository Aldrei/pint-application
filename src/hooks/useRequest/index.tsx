import * as React from 'react';

import axios from '../../hooks/useConfigAxios';

// import { propertiesVideosService } from '../../services/properties';

import { IServiceRequestStatus } from '../../types';

interface IProps {
  type: 'propertyVideoDelete';
  code: string;
  videoId: string;
}

const useRequest = ({ code, videoId }: IProps) => {
  const [status, setStatus] = React.useState<IServiceRequestStatus>('idle');

  axios.get('https://swapi.dev/api/people', {
    onDownloadProgress: progressEvent => {
      setStatus('loading');
    }
  })
    .then((res) => {
      console.log('DEBUG RESPONSE:', res);
      setStatus('success');
    })
    .catch((err) => {
      console.log('DEBUG ERROR:', err);
      setStatus('failed');
    });

  return { status };
};

export default useRequest;