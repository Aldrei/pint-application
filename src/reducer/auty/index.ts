import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { authService, IAuthServiceAccessTokenRequest, IAuthServiceAccessTokenResponse } from '../../services/auth';

import { IServiceRequest } from '../../types';

export interface IAutyState extends IServiceRequest {
  whoIsAuth: object;
  accessToken: IAuthServiceAccessTokenResponse;
}

const initialState: IAutyState = {
  status: 'idle',
  whoIsAuth: {},
  accessToken: {} as IAuthServiceAccessTokenResponse,
};

// The function below is called a thunk and allows us to perform async logic. It
// can be dispatched like a regular action: `dispatch(incrementAsync(10))`. This
// will call the thunk with the `dispatch` function as the first argument. Async
// code can then be executed and other actions can be dispatched. Thunks are
// typically used to make async requests.
export const authServiceThunk = createAsyncThunk(
  'counter/fetchCount',
  async (data: IAuthServiceAccessTokenRequest) => {
    const response = await authService.accessToken(data);
    // The value we return becomes the `fulfilled` action payload
    return response;
  }
);

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {},
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      .addCase(authServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(authServiceThunk.fulfilled, (state, action) => {
        console.log('authServiceThunk fulfilled:', action);
        state.status = 'success';
        state.accessToken = action.payload?.data?.access_token ? action.payload.data : {};
        state.data = action.payload.data;
      })
      .addCase(authServiceThunk.rejected, (state, action) => {
        console.log('authServiceThunk rejected:', action);
        state.status = 'failed';
        state.data = action.payload as object;
        state.accessToken = {};
      });
  },
});

// The function below is called a selector and allows us to select a value from
// the state. Selectors can also be defined inline where they're used instead of
// in the slice file. For example: `useSelector((state: RootState) => state.counter.value)`
// export const selectCount = (state: RootState) => state.counter.value;

export default authSlice.reducer;
