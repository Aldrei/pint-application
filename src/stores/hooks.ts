import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import type { RootState, AppDispatch } from './index';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const useSelectorCallback = (state: RootState, reducer: keyof RootState) => state[reducer];
