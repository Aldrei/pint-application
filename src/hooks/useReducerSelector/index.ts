import { RootState, IReducersType } from '../../store';
import { useAppSelector } from '../../store/hooks';

export const useAppSelectorBlaBlaBal = (reducer: keyof IReducersType) => useAppSelector((state: RootState) => state[reducer]);