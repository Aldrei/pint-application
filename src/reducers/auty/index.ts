import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { authService, IAuthServiceAccessTokenRequest, IAuthServiceAccessTokenResponse, IAuthServiceRevokeRequest } from '../../services/auth';
import { AppThunk } from '../../stores';

import { IServiceRequest, IWhoIsAuth } from '../../types';

export interface IAutyState extends IServiceRequest {
  whoIsAuth: IWhoIsAuth;
  accessToken: IAuthServiceAccessTokenResponse;
}

const initialState: IAutyState = {
  name: 'authReducer',
  status: 'idle',
  whoIsAuth: {} as IWhoIsAuth,
  accessToken: {} as IAuthServiceAccessTokenResponse,
};

export const authServiceThunk = createAsyncThunk(
  'auth/access_token',
  async (data: IAuthServiceAccessTokenRequest) => {
    const response = await authService.accessToken(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const revokeServiceThunk = createAsyncThunk(
  'auth/revoke',
  async (data: IAuthServiceRevokeRequest) => {
    const response = await authService.revoke(data);
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const whoServiceThunk = createAsyncThunk(
  'auth/who_is_auth',
  async () => {
    const response = await authService.who();
    // The value we return becomes the `fulfilled` action payload

    return response;
  }
);

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  // The `reducers` field lets us define reducers and generate associated actions
  reducers: {
    clearAccessToken: (state) => {
      state.accessToken = {} as IAuthServiceAccessTokenResponse;
      state.whoIsAuth = {} as IWhoIsAuth;
    }
  },
  // The `extraReducers` field lets the slice handle actions defined elsewhere,
  // including actions generated by createAsyncThunk or in other slices.
  extraReducers: (builder) => {
    builder
      /** Access token */
      .addCase(authServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(authServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.accessToken = action.payload?.data?.access_token ? action.payload.data : {};
        state.data = action.payload.data;
      })
      .addCase(authServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
        state.accessToken = {};
      })
      /** Revoke. */
      .addCase(revokeServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(revokeServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.data = action.payload.data;
        state.accessToken = {};
        state.whoIsAuth = {} as IWhoIsAuth;
      })
      .addCase(revokeServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.data = action.payload as object;
      })
      /** Who is auth. */
      .addCase(whoServiceThunk.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(whoServiceThunk.fulfilled, (state, action) => {
        state.status = 'success';
        state.whoIsAuth = action.payload.data;
      })
      .addCase(whoServiceThunk.rejected, (state, action) => {
        state.status = 'failed';
        state.whoIsAuth = action.payload as IWhoIsAuth;
      });
  },
});

const { clearAccessToken } = authSlice.actions;

export const clearAccessTokenReducer = (): AppThunk => (dispatch) => {
  dispatch(clearAccessToken());
};

export default authSlice.reducer;
