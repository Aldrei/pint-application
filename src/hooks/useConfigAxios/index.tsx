import axios from 'axios';

import { clearAccessTokenReducer } from '../../reducers/auty';
import { useAppDispatch } from '../../stores/hooks';

const api = axios.create({
  baseURL: process.env.REACT_APP_API_BASE_URL,
});

const getToken = () => {
  try {
    const persistImob = localStorage.getItem('persist:imob');
    if (persistImob) {
      const persistImobData = JSON.parse(persistImob);

      if (persistImobData.authReducer) {
        const authReducerData = Object.assign({}, JSON.parse(persistImobData.authReducer));
        return authReducerData?.accessToken?.access_token;
      }
    }

    return '';
  } catch (error) {
    console.log(error);
    return '';
  }
};


// Add a request interceptor
api.interceptors.request.use(function (config) {
  if (config.headers) config.headers['Authorization'] = `Bearer ${getToken()}`;
  return config;
}, function (error) {
  // Do something with request error
  return Promise.reject(error);
});

export const useConfigAxios = () => {
  const dispatch = useAppDispatch();

  // Add a response interceptor
  api.interceptors.response.use((response) => {
    console.log('api.interceptors.response success...');
    // Any status code that lie within the range of 2xx cause this function to trigger
    /** NOTE: Suppress 'A non-serializable value was detected in the state, in the path' error */
    return JSON.parse(JSON.stringify(response));
  }, (error) => {
    console.log('api.interceptors.response error...', { error });
    // Any status codes that falls outside the range of 2xx cause this function to trigger
    // return Promise.reject(error);

    if (error.response.status === 401) {
      dispatch(clearAccessTokenReducer());
    }

    /** NOTE: Suppress 'A non-serializable value was detected in the state, in the path' error */
    return JSON.parse(JSON.stringify(error.response));
  });

  return [];
};

export default api;