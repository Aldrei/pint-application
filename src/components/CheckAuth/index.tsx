import React, { useEffect, useState } from 'react';

import { Navigate } from 'react-router-dom';

import { IAutyState, whoServiceThunk } from '../../reducers/auty';
import { useAppSelectorBlaBlaBal } from '../../hooks/useReducerSelector';
import { useAppDispatch } from '../../hooks/useReducerDispatch';

import { useConfigAxios } from '../../hooks/useConfigAxios';

import Menu from '../Menu';
import Header from '../Header';

const CheckAuth = ({ children }: { children: JSX.Element }) => {
  const dispatch = useAppDispatch();

  const { accessToken } = useAppSelectorBlaBlaBal('authReducer') as IAutyState;
  const [menuActive, setMenuActive] = useState(false);

  useConfigAxios();

  useEffect(() => {
    /**
     * Soon after user loggin and is redirect to dashboard page, getToke() don't can get the new token from localStorage.
     * So, pass the new token by parameter.
    */
    if (accessToken?.access_token) dispatch(whoServiceThunk(accessToken?.access_token));
  }, [accessToken]);

  const toggleDrawer =
    (
      // event: React.KeyboardEvent | 
      event: React.MouseEvent, open: boolean) => {
      
      // if (
      //   event.type === 'keydown' &&
      //   ((event as React.KeyboardEvent).key === 'Tab' ||
      //     (event as React.KeyboardEvent).key === 'Shift')
      // ) {
      //   return;
      // }

      setMenuActive(open);
    };

  if (!accessToken?.access_token) 
    return <Navigate to="/login" />;

  return (
    <React.Fragment>
      <Header toggleMenu={(e, action) => toggleDrawer(e, action)} />
      <Menu
        handleClose={() => {
          setMenuActive(false);
        }}
        menuActive={menuActive} />
      {children}
    </React.Fragment>
  );
};

export default CheckAuth;
