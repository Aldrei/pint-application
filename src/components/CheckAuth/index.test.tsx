import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import { useAppSelectorBlaBlaBal } from '../../hooks/useReducerSelector';
import { IAutyState } from '../../reducer/auty';

import CheckAuth from './index';

jest.mock('../../hooks/useReducerSelector', () => ({
  useAppSelectorBlaBlaBal: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  Navigate: jest.fn(),
  useNavigate: jest.fn()
}));

jest.mock('../../store/hooks', () => ({
  useAppDispatch: jest.fn(),
}));

describe('CheckAuth component', () => {
  const useAppSelectorBlaBlaBalMocked = useAppSelectorBlaBlaBal as jest.MockedFunction<typeof useAppSelectorBlaBlaBal>;

  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('Should render children correctly', () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: 'dummy-token' } } as IAutyState);
    const nodeEl = render(<CheckAuth><div>Children element</div></CheckAuth>);
    expect(nodeEl.baseElement).toHaveTextContent('Children element');
    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should redirect to Login page', () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: null } } as unknown as IAutyState);
    const nodeEl = render(<CheckAuth><div>Children element</div></CheckAuth>);
    expect(nodeEl.baseElement).not.toHaveTextContent('Children element');
    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  it('Should open side left menu', () => {
    useAppSelectorBlaBlaBalMocked.mockReturnValue({ accessToken: { access_token: 'dummy-token' } } as IAutyState);
    
    const nodeEl = render(<CheckAuth><div>Children element</div></CheckAuth>);
    expect(nodeEl.baseElement).toHaveTextContent('Children element');

    const leftMenuButton = nodeEl.getByTestId('toggle-left-menu-button');
    fireEvent.click(leftMenuButton);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });
});
