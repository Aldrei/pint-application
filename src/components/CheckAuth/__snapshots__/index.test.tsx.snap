// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckAuth component Should open side left menu 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          style="flex-direction: row; background-color: rgb(19, 47, 76); border: 1px solid rgb(23, 58, 94);"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
            data-testid="toggle-left-menu-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
          >
            MUI
          </div>
          <div
            class="css-1orzy9s"
          >
            <div
              class="css-1iz4iln"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                placeholder="Searchâ€¦"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiBox-root css-5nwj3y"
          >
            <button
              aria-haspopup="true"
              aria-label="Account settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="account-menu-button-open-testid"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
              >
                M
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div />
    <div>
      Children element
    </div>
  </div>
  <div
    class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1u2w381-MuiModal-root-MuiDrawer-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-4t3x6l-MuiPaper-root-MuiDrawer-paper"
      style="background-color: rgb(19, 47, 76); border-right: 1px solid rgb(23, 58, 94); webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiBox-root css-1hskriy"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MoveToInboxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Inbox
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Starred
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MoveToInboxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Send email
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Drafts
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
        </ul>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MoveToInboxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  All mail
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Trash
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              style="flex-direction: row; color: rgb(178, 186, 194);"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                style="color: rgb(178, 186, 194);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MoveToInboxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Spam
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`CheckAuth component Should redirect to Login page 1`] = `
<body>
  <div />
</body>
`;

exports[`CheckAuth component Should render children correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          style="flex-direction: row; background-color: rgb(19, 47, 76); border: 1px solid rgb(23, 58, 94);"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
            data-testid="toggle-left-menu-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
          >
            MUI
          </div>
          <div
            class="css-1orzy9s"
          >
            <div
              class="css-1iz4iln"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                placeholder="Searchâ€¦"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiBox-root css-5nwj3y"
          >
            <button
              aria-haspopup="true"
              aria-label="Account settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="account-menu-button-open-testid"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-azeo8y-MuiAvatar-root"
              >
                M
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div />
    <div>
      Children element
    </div>
  </div>
</body>
`;
